### ISSUE TITLE
How to configure the project

### ISSUE THREAD

**Issue by yanko** [JoeyDeVries/Cell#3]
is there a step by step how to compile, run, required libs....
i have spent some time now trying to compile the code and been having issues with assimp and zilb.

tx.

**Comment by JoeyDeVries** [comment]
If memory serves me right Assimp (and ZLib) have been built statically (so with the static flags set during the CMake configuration). I believe I also had to enable some option(s) during Assimp's CMake configuration stage to include or custom built ZLib so I could get the static thing to work.

**Comment by iKlsR** [comment]
@JoeyDeVries Please consider moving to a cmake based build, this is pretty cumbersome on a new machine and the assimp build step is pretty disjoint from everything else. I'm loving this project otherwise, amazing job, looking forward to seeing more.

**Comment by JoeyDeVries** [comment]
Yeah, this project so far has been a bit of an experiment for myself; wasn't really meant to be used by a lot of people (yet) so I haven't taken the time yet so set up a proper build system. I'll have to take a look soon to see if I can  get the Assimp static build working w/ CMake.

**Comment by urzq** [comment]
Hello there!

I'm learning CMake for my projects (see a playground here: https://github.com/urzq/TinySubMake), and I'm wondering if you would be interested by a better CMakeList for the cell engine :) ? 

We could have all the external projects (GLFW, glad, ASSIMP) as submodules, and CMake would build static version of them. There is also IMGUI and STB, but I'm not sure how to handle them yet, because they don't seem to play nicely with cmake...

Basically, the plan is to have a very simple build process for the users, 

Anyway tell me if you are interested or not, and we can see if I have enough motivation / skill to do that :D (I need to learn pull request too).

**Comment by JoeyDeVries** [comment]
Definitely! Having a proper CMake module for Cell would be awesome :) 

For STB_Image (and GLAD)  which were separate cpp/header files instead of a library I remember making a project out of the individual files (with CMake) and then linking those with each example in the LearnOpenGL repository (https://github.com/JoeyDeVries/LearnOpenGL/blob/master/CMakeLists.txt).
